### PR TITLE
fix: fix anchors in changelog documentation

### PR DESCRIPTION
**Pull request by JCHacking**
### Description
Fixed the problem where the links for the changelog in the web documentation were not being generated correctly

### Addressed Issue
#2094 

### Checklist

- [X] I have read and understand the [contributing guidelines](../CONTRIBUTING.md#pull-requests)
- [X] This PR fixes a defect, and I have provided tests to verify that the fix is effective
- [ ] This PR implements an enhancement, and I have provided tests to verify that it works as intended
- [ ] This PR introduces changes to the database model, and I have added corresponding [update logic](https://github.com/DependencyTrack/dependency-track/tree/master/src/main/java/org/dependencytrack/upgrade)
- [ ] This PR introduces new or alters existing behavior, and I have updated the [documentation](https://github.com/DependencyTrack/dependency-track/tree/master/docs/_docs) accordingly

### CODE REVIEW

**Comment by JCHacking** [comment]
As far as I can see it was failing because it always caught the id of the last h6 (Software Bill of Materials).

If we set the ids to be generated, it works in the versions, but in the files it still fails.

**Comment by nscuro** [comment]
Thanks for looking into this!

> Where are these links generated?

They are generated by https://github.com/allejo/jekyll-anchor-headings which is located here in the repo: https://github.com/DependencyTrack/dependency-track/blob/master/docs/_includes/anchor_headings.html

To be honest I added it a while ago due to a lack of better options - before adding it there were no anchors at all. I'm saying this because there *may* be better options out there that do the job more reliably, so please feel free to experiment with other options as you see fit.

**Comment by JCHacking** [comment]
I have checked the tool and it seems that when in a markdown there is a h1 - h6 it automatically generates an id, which is translated into an anchor. So with archives there is no automatic solution, the only one would be to generate the markdown from _post to add the link.

**Comment by JCHacking** [comment]
Hi @nscuro 
After a thorough review, I don't see a better way to do all this, this is the best that can be achieved with Jekyll.

Do you want to put links to the files of each release? For this we would have to modify the _post, they are generated automatically? 

If so, I can edit the existing ones and make the new ones incorporate it, as long as you are ok with it.

**Comment by nscuro** [comment]
Thanks for investigating!

> Do you want to put links to the files of each release? For this we would have to modify the _post, they are generated automatically?

If specifying the links manually fixes the issue, then I'd say let's go with that. It may be tedious, but IMO the important part is that they *work* :D

Would be much appreciated if you could do that!

**Comment by JCHacking** [comment]
Hi @nscuro 
Modified all _posts with a script (Sorry so many commits, you can see better from github the differences than from git by console)

File: docs/_posts/2024-07-08-v4.11.5.md [hunk 1]
```diff
@@ -30,21 +30,21 @@ We thank all organizations and individuals who contributed to this release, from
...
-###### dependency-track-bundled.jar
+###### [dependency-track-bundled.jar](https://github.com/DependencyTrack/dependency-track/releases/download/4.11.5/dependency-track-bundled.jar)
 
 | Algorithm | Checksum                                                         |
 |:----------|:-----------------------------------------------------------------|
 | SHA-1     | eba6cbaa6c2da9ffb295da83ed39af68ff4130a8                         |
 | SHA-256   | 7ebb11573b2a59084ed98fe92d363240c910dc7b5aa7ebeda64bee7d47089d9a |
 
-###### frontend-dist.zip
+###### [frontend-dist.zip](https://github.com/DependencyTrack/dependency-track/releases/download/4.11.5/frontend-dist.zip)
```
Comment (nscuro) [inline docs/_posts/2024-07-08-v4.11.5.md:47]: Frontend links should point to the `DependencyTrack/frontend` repository. The current links yield a 404.

```suggestion
###### [frontend-dist.zip](https://github.com/DependencyTrack/frontend/releases/download/4.11.5/frontend-dist.zip)
```

**Review by nscuro — CHANGES_REQUESTED** [review]

**Comment by JCHacking** [comment]
I didn't notice that case, fixed in all files

**Review by nscuro — APPROVED** [review]
Thanks!

**Pull request merged**